src/ggml-cuda/fattn-vec-f16.cuh

```diff
@@ -220,7 +220,8 @@ static __global__ void flash_attn_vec_ext_f16(
         for (int j = 0; j < ncols; ++j) {
             half kqmax_new_j = ncols == 1 ? kqmax_new : kqmax_new_arr[j];
 
-            kqmax_new_j = warp_reduce_max(kqmax_new_j);
+            /* kqmax_new_j in every thread within warp is same after operate at line 199,this reduce can be omit  */
```

Comment:
Strictly speaking it is not exactly the same since the order in which you sum up floating point values makes a difference for the result but this is negligible. And there is no reason why the maximum value in particular should be preferred.